docs/classic-ui/portlets.md

```diff
@@ -1,13 +1,185 @@
+    """Portlet renderer."""
+
+    # Define the page template that will be used to render the portlet
+    template = ViewPageTemplateFile("my_portlet.pt")
+
+    def message(self):
+        """This method is called by the page template to render the portlet."""
+        return self.data.message
+
+    def render(self):
+        """This method is called whenever the portlet is rendered."""
+        return self.template()
+
+
+class AddForm(base.AddForm):
+    """Portlet add form."""
+
+    schema = IExamplePortlet
+    label = "Add Greeting Portlet"
+    description = "This portlet displays a greeting."
+
+    def create(self, data):
+        return Assignment(**data)
+
+
+class EditForm(base.EditForm):
+    """Portlet edit form."""
+
+    schema = IExamplePortlet
+    label = "Edit Greeting Portlet"
+    description = "This portlet displays a greeting."
+
+```
+
+Here is the example for a simple page template ``my_portlet.pt``:
+
+```HTML
+<html
+  xmlns="http://www.w3.org/1999/xhtml"
+  xmlns:tal="http://xml.zope.org/namespaces/tal"
+  xmlns:i18n="http://xml.zope.org/namespaces/i18n"
+  tal:omit-tag=""
+>
+  <div class="card portlet" i18n:domain="my-example">
+    <div class="card-header">Greeter</div>
+
+    <div class="card-body">
+      <p>${python:view.message()}</p>
+    </div>
+
+    <div class="card-footer portletFooter"></div>
+  </div>
+</html>
+```
+
+This portlet class defines a portlet with the title "My Portlet" and the name "my-portlet". When the portlet is rendered, it will use the template file `my_portlet.pt` to generate its HTML output that displays a greeting message.
+
+The message is set when the portlet is added and can be edited in the portlet's edit form.
+
+To register this portlet with Plone, you will need to create a ``configure.zcml`` file that tells Plone about the portlet, and - after a restart - you can add it to a Plone page using the "manage portlets" screen.
+Here is an example ``configure.zcml`` file that registers the MyPortlet class defined above:
+
+```XML
+<configure xmlns="http://namespaces.zope.org/zope"
+           xmlns:browser="http://namespaces.zope.org/browser"
+           xmlns:plone="http://namespaces.plone.org/plone"
+           i18n_domain="example.portlet">
+
+  <plone:portlet
+      title="Example Portlet"
+      description="A portlet that displays a greeting message"
+      addview="example.portlet.add"
+      editview="example.portlet.edit"
+      assignment="example.portlet.Assignment"
+      renderer="example.portlet.Renderer"
+      schema="example.portlet.IExamplePortlet"
+  />
+
+</configure>
+```
+
+This file registers a portlet with the following properties:
+
+- *Title:* Example" Portlet"
+- *Description:* "A portlet that displays a greeting message"
+- *Add form:* "example.portlet.add"
+- *Edit form:* "example.portlet.edit"
+- *Assignment:* "example.portlet.Assignment"
+- *Renderer:* "example.portlet.Renderer"
+- *Schema:* "example.portlet.IExamplePortlet"
```

Comment:
The above bullet list should be converted to a [MyST definition list](https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#definition-lists).

```
Title
:   `Example Portlet`
```

...and so on. Note that the definitions of the terms are inline literals.

Reply:
Yeah, legibility for the reader is now improved. https://deploy-preview-1369--6-dev-docs-plone-org.netlify.app/classic-ui/portlets.html#writing-a-custom-portlet